src/main/java/com/aws/greengrass/deployment/KernelUpdateDeploymentTask.java

```diff
@@ -94,10 +101,12 @@ public DeploymentResult call() {
             logger.atError("deployment-errored", e).log();
             if (KERNEL_ACTIVATION.equals(stage)) {
                 try {
-                    deployment.setDeploymentStage(KERNEL_ROLLBACK);
+                    final boolean bootstrapOnRollbackRequired = kernelAlternatives.prepareBootstrapOnRollbackIfNeeded(
```

Comment:
customer's deployment directory did not have any `rollback_boostrap_task.txt` file, so will this even trigger in their case?

Reply:
Please update the description with how this change was tested to validate it works.

Reply:
`prepareBootstrapOnRollbackIfNeeded` determines if bootstrap is required from the rollback config snapshot and not the `rollback_boostrap_task.txt` file. Rollback bootstrap task list is persisted in this method which is later used by the kernel (after reboot) to determine deployment stage. And that should be accurate.

Reply:
Updated description

Reply:
The UAT does not reboot the device, please test manually using the exact customer scenario including reboot.

Reply:
Just validated the behavior with manual testing. Deploying components from the exact customer scenario goes through the rollback bootstrap deployment stage when rollback is triggered after component startup lifecycle errors.